LiteCore/Query/SQLiteKeyStore+ArrayIndexes.cc

```diff
@@ -31,21 +31,23 @@ namespace litecore {
                     createUnnestedTable(itPath.value(), plainTableName, unnestTableName);
         }
         Array::iterator iExprs(spec.what());
-        return createIndex(spec, unnestTableName, iExprs);
+        return createIndex(spec, plainTableName, iExprs);
     }
 
     std::pair<string, string> SQLiteKeyStore::createUnnestedTable(const Value* expression, string plainParentTable,
                                                                   string parentTable) {
         // Derive the table name from the expression it unnests:
         if ( plainParentTable.empty() ) plainParentTable = parentTable = tableName();
         QueryParser qp(db(), "", plainParentTable);
-        auto [plainTableName, unnestTableName] = qp.unnestedTableName(expression);
+        string      plainTableName    = qp.unnestedTableName(expression);
+        string      unnestTableName   = KeyStore::hexName(plainTableName);
+        string      quotedParentTable = CONCAT(sqlIdentifier(parentTable));
 
         // Create the index table, unless an identical one already exists:
         string sql = CONCAT("CREATE TABLE " << sqlIdentifier(unnestTableName)
                                             << " "
                                                "(docid INTEGER NOT NULL REFERENCES "
-                                            << sqlIdentifier(parentTable)
+                                            << quotedParentTable
```

Comment:
This is actually better the other way -- sqlIdentifier() doesn't allocate a string.